src/test/java/itest/ReportIT.java

```diff
@@ -0,0 +1,160 @@
+                                    postResponse.complete(null);
+                                }
+                            });
+
+            postResponse.get();
+
+            // Save the recording to archive
+            webClient
+                    .patch(String.format("%s/%s", RECORDING_REQ_URL, TEST_RECORDING_NAME))
+                    .sendBuffer(
+                            Buffer.buffer("SAVE"),
+                            ar -> {
+                                if (assertRequestStatus(ar, savedRecordingName)) {
+                                    MatcherAssert.assertThat(
+                                            ar.result().statusCode(), Matchers.equalTo(200));
+                                    MatcherAssert.assertThat(
+                                            ar.result()
+                                                    .getHeader(HttpHeaders.CONTENT_TYPE.toString()),
+                                            Matchers.equalTo(HttpMimeType.PLAINTEXT.mime()));
+                                    savedRecordingName.complete(ar.result().bodyAsString());
+                                }
+                            });
+
+            savedRecordingName.get();
+
+            // Get a report for the above recording
+            CompletableFuture<Buffer> getResponse = new CompletableFuture<>();
+            webClient
+                    .get(String.format("%s/%s", REPORT_REQ_URL, savedRecordingName.get()))
+                    .send(
+                            ar -> {
+                                if (assertRequestStatus(ar, getResponse)) {
+                                    MatcherAssert.assertThat(
+                                            ar.result().statusCode(), Matchers.equalTo(200));
+                                    MatcherAssert.assertThat(
+                                            ar.result()
+                                                    .getHeader(HttpHeaders.CONTENT_TYPE.toString()),
+                                            Matchers.equalTo(HttpMimeType.HTML.mime()));
+                                    getResponse.complete(ar.result().bodyAsBuffer());
+                                }
+                            });
+
+            getResponse.get();
+
+        } finally {
+            // Clean up recording
+            CompletableFuture<JsonObject> deleteResponse = new CompletableFuture<>();
+            webClient
+                    .delete(String.format("/api/v1/recordings/%s", savedRecordingName.get()))
+                    .send(
+                            ar -> {
+                                if (assertRequestStatus(ar, deleteResponse)) {
+                                    deleteResponse.complete(ar.result().bodyAsJsonObject());
+                                }
+                            });
+
+            MatcherAssert.assertThat(deleteResponse.get(), Matchers.equalTo(null));
```

Comment:
I think this cleanup should delete the active and archived recordings as well, right?

Reply:
Both recordings should be cleaned up now